resources/module_3_1/viewer-request-function.js

```diff
@@ -0,0 +1,25 @@
+import cf from 'cloudfront';
+const kvsId = '__KVS_ID__'; // replaced during CDK template synthesis
+const kvsHandle = cf.kvs(kvsId);
+
+async function handler(event) {
+    const request = event.request;
+    const configRaw = await kvsHandle.get('config'); // CFF runtime doesn't support nested await
+    const config = JSON.parse(configRaw);
+
+    // Check if site was already visited by the user
+    const returningUser = Boolean(request.cookies['X-Experiment']);
+    console.log('X_Experiment_U: ' + (returningUser ? 'RETURNING USER' : 'NEW USER'));
+
+    // If already visited, parse the experiment value, else generate one
+    const experimentValue = returningUser ? parseInt(request.cookies["X-Experiment"].value) : Math.floor(Math.random() * 100);
+    console.log('EXPERIMENT VALUE: ' + experimentValue);
+
+    // Set the correct URI for the experiment
+    request.uri = experimentValue > config.THRESHOLD ? config.URI_A : config.URI_B;
+    console.log('X_Experiment_V: ' + request.uri);
```

Comment:
You should log something like `console.log("X_Experiment_V " + (request.uri == '/index.html' ? 'A_VERSION' : 'B_VERSION'));` and make sure that the dashboard still works.

Reply:
Fixed in https://github.com/aws-samples/ab-testing-at-edge/pull/11/commits/c443a190a0dd303a7c015dcde479c12b736e838c